internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -222,6 +226,114 @@ ExecExprVisitor::ExecUnaryRangeVisitorDispatcher(UnaryRangeExpr& expr_raw) -> Bi
 }
 #pragma clang diagnostic pop
 
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "Simplify"
+template <typename T>
+auto
+ExecExprVisitor::ExecBinaryArithOpEvalRangeVisitorDispatcher(BinaryArithOpEvalRangeExpr& expr_raw) -> BitsetType {
+    auto& expr = static_cast<BinaryArithOpEvalRangeExprImpl<T>&>(expr_raw);
+    using Index = scalar::ScalarIndex<T>;
+    auto arith_op = expr.arith_op_;
+    auto right_operand = expr.right_operand_;
+    auto op = expr.op_type_;
+    auto val = expr.value_;
+
+    switch (op) {
+        case OpType::Equal: {
+            switch (arith_op) {
+                case ArithOpType::Add: {
+                    auto index_func = [val, right_operand, arith_op](Index* index) {
```

Comment:
As above mentioned, I suggest that we simply use raw data to do brute-force search here. You can abstract a newly function like `ExecRangeVisitorImpl`, this function accepts a `ElemFunc` and will apply the `ElemFunc` for every row in every chunk in the segment.

Reply:
I implemented an `ExexDataRangeVisitorImpl` that does the filtering of the rows. It passed the tests, but I'd like to ask if the changes I made are what you had in mind, too. :)

Reply:
Yes! Well-matched what I wanted. thx.